tool/src/java/org/sakaiproject/gradebookng/tool/panels/CategoryColumnCellPanel.java

```diff
@@ -44,9 +46,30 @@ public void onInitialize() {
 		Map<String,Object> modelData = (Map<String,Object>) this.model.getObject();
 		
 		Double score = (Double) modelData.get("score");		
+		final String studentUuid = (String) modelData.get("studentUuid");
+		final Long categoryId = (Long) modelData.get("categoryId");
 						
 		//score label
-		add(new Label("score", getCategoryScore(score)));
+		Label scoreLabel = new Label("score", Model.of(getCategoryScore(score))) {
+			@Override
+			public void onEvent(IEvent<?> event) {
+				super.onEvent(event);
+				if (event.getPayload() instanceof ScoreChangedEvent) {
+					ScoreChangedEvent scoreChangedEvent = (ScoreChangedEvent) event.getPayload();
+					if (studentUuid.equals(scoreChangedEvent.getStudentUuid()) && 
+							categoryId.equals(scoreChangedEvent.getCategoryId())) {
+
+						GbStudentGradeInfo studentGradeInfo = businessService.buildGradeMatrix(businessService.getGradebookAssignments(), Arrays.asList(new String[]{studentUuid}), null, null).get(0);
```

Comment:
Ah yep.. as mentioned on the team call, this is the call to the service to pull back the latest category average via `buildGradeMatrix`. Perhaps would be nicer if there was a more targeted API call?